src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs

```diff
@@ -280,40 +211,30 @@ public void QueueTimelineRecordUpdate(Guid timelineId, TimelineRecord timelineRe
         {
             while (!_jobCompletionSource.Task.IsCompleted || runOnce)
             {
-                if (!runOnce)
+                List<List<string>> batchedLines = new List<List<string>>();
+                List<string> currentBatch = new List<string>();
+                string line;
+                while (_webConsoleLineQueue.TryDequeue(out line))
                 {
-                    bool forceProcess = await _webConsoleLineQueueSemaphore.WaitAsync(_delayForWebConsoleLineDequeue);
-                    if (forceProcess)
+                    currentBatch.Add(line);
+                    // choose 100 lines since the whole web console UI will only shows about 40 lines in a 15" monitor.
+                    if (currentBatch.Count > 100)
                     {
-                        Trace.Verbose("Process web console line queue since the force process signal been fired.");
+                        batchedLines.Add(currentBatch.ToList());
+                        currentBatch.Clear();
                     }
-                    else
+
+                    // process at most about 500 lines of web console line during regular timer dequeue task.
```

Comment:
Since we only keep 1000 lines in the web UI buffer, we could even throw away batches here if it's more than 1000 lines :)